CLAUDE_PLUGIN_ROOT not injected into plugin hooks — causes hook errors on every tool call

## Bug 1: \`CLAUDE_PLUGIN_ROOT\` empty string causes all plugin hooks to fail

### Environment
- Claude Code v2.1.89
- Plugins: \`claude-subconscious\` v2.0.2, \`episodic-memory\` v1.0.15
- OS: Linux

### Symptoms

Every tool call shows hook errors in the Claude Code UI:
```
PreToolUse:Glob hook error
PreToolUse:Bash hook error
UserPromptSubmit hook error
SessionStart:startup hook error
```

### Root Cause

Plugin hooks use `${CLAUDE_PLUGIN_ROOT}` in their commands (from `hooks.json`):
```json
"command": "node \"${CLAUDE_PLUGIN_ROOT}/hooks/silent-npx.cjs\" tsx \"${CLAUDE_PLUGIN_ROOT}/scripts/pretool_sync.ts\""
```

At runtime, `CLAUDE_PLUGIN_ROOT` is **empty string** in the hook subprocess environment. Confirmed via `env | grep CLAUDE_PLUGIN_ROOT` inside a Bash tool call — returns `CLAUDE_PLUGIN_ROOT=''`.

This causes:
```
Error: Cannot find module '/hooks/silent-npx.cjs'
  code: 'MODULE_NOT_FOUND'
Exit: 1  →  Claude Code shows "hook error"
```

**Reproduce:**
```bash
# Confirm CLAUDE_PLUGIN_ROOT is empty in CC environment
echo "CLAUDE_PLUGIN_ROOT='${CLAUDE_PLUGIN_ROOT}'"
# → CLAUDE_PLUGIN_ROOT=''

# Reproduce the failure exactly as CC runs it
PLUGIN_ROOT=~/.claude/plugins/cache/claude-subconscious/claude-subconscious/2.0.2
CLAUDE_PLUGIN_ROOT='' bash -c 'node "${CLAUDE_PLUGIN_ROOT}/hooks/silent-npx.cjs" tsx "${CLAUDE_PLUGIN_ROOT}/scripts/pretool_sync.ts"'
# → MODULE_NOT_FOUND, exit 1

# Works fine when CLAUDE_PLUGIN_ROOT is correctly set:
CLAUDE_PLUGIN_ROOT="$PLUGIN_ROOT" bash -c 'node "${CLAUDE_PLUGIN_ROOT}/hooks/silent-npx.cjs" tsx "${CLAUDE_PLUGIN_ROOT}/scripts/pretool_sync.ts"'
# → exit 0
```

Affects all plugins whose `hooks.json` commands reference `${CLAUDE_PLUGIN_ROOT}`:
- `claude-subconscious`: `pretool_sync.ts` (fires on **every** PreToolUse), `sync_letta_memory.ts`, `session_start.ts`
- `episodic-memory`: `sync --background` on SessionStart

---

## Bug 2: Startup race condition with external services

Plugin hooks connecting to Letta (`127.0.0.1:8283`) fail on session start if the container isn't fully ready yet. The `session_start.ts` hook has a hard **5-second timeout** but service startup can take longer, causing:
```
SessionStart:startup hook error  (×2)
```
This happens even when the Docker container shows `Up N minutes` — the HTTP API may still be initializing.

---

## Fixes

### Fix 1 — `CLAUDE_PLUGIN_ROOT` injection (CC-side fix needed)

Claude Code should inject `CLAUDE_PLUGIN_ROOT` set to the plugin's cache directory when spawning each plugin hook subprocess.

**Workaround until fixed** — patch `hooks.json` to hardcode the absolute path:
```bash
PLUGIN_ROOT=~/.claude/plugins/cache/claude-subconscious/claude-subconscious/2.0.2
sed -i "s|\${CLAUDE_PLUGIN_ROOT}|$PLUGIN_ROOT|g" "$PLUGIN_ROOT/hooks/hooks.json"
```
> ⚠️ Re-apply after each plugin update.

Or disable the affected plugin entirely:
```bash
claude plugin disable claude-subconscious
```

### Fix 2 — Startup race condition (plugin-side fix)

Add a health-check with retry loop before connecting to external services, or mark startup hooks `async: true` so timeout failures don't surface as visible errors:
```json
{ "type": "command", "command": "...", "timeout": 15, "async": true }
```

--- a/lib/12-plugins.sh
+++ b/lib/12-plugins.sh
@@ -67,6 +67,18 @@ else
           warn "subconscious: npm install failed — hooks may not work"
       fi
 
+      # Patch hooks.json: replace ${CLAUDE_PLUGIN_ROOT} with hardcoded absolute path.
+      # CC does not inject CLAUDE_PLUGIN_ROOT into hook subprocesses — empty string causes
+      # "Cannot find module '/hooks/...'" (MODULE_NOT_FOUND) on every hook invocation.
+      # Guard: only apply when literal '${CLAUDE_PLUGIN_ROOT}' is still present (idempotent).
+      _SUBCON_HOOKS="$_SUBCON_DIR/hooks/hooks.json"
+      if [[ -n "$_SUBCON_DIR" ]] && [[ -f "$_SUBCON_HOOKS" ]] &&
+          grep -q '\${CLAUDE_PLUGIN_ROOT}' "$_SUBCON_HOOKS" 2>/dev/null; then
+        sed -i "s|\${CLAUDE_PLUGIN_ROOT}|${_SUBCON_DIR}|g" "$_SUBCON_HOOKS" &&
+          ok "subconscious: patched hooks.json (CLAUDE_PLUGIN_ROOT → hardcoded path)" ||
+          warn "subconscious: hooks.json CLAUDE_PLUGIN_ROOT patch failed"
+      fi
+
       # Patch /dev/tty crash: createWriteStream('/dev/tty') throws unhandled ENXIO in hook context
       # (no terminal). Add .on("error") handler so the stream fails gracefully instead of crashing.
       _SUBCON_SESS="$_SUBCON_DIR/scripts/session_start.ts"

--- a/test/session-review.bats
+++ b/test/session-review.bats
@@ -842,3 +842,24 @@ setup() {
 @test "ROT: assembled titan-setup.sh includes password rotation logic" {
   grep -q '_RUNNING_PASS=$(docker inspect letta-server' "$REPO/titan-setup.sh"
 }
+
+
+# ── CLAUDE_PLUGIN_ROOT injection fix (issue #67, 2026-04-01) ──
+@test "I67: lib/12 patches hooks.json to replace CLAUDE_PLUGIN_ROOT with hardcoded path" {
+  grep -q 'CLAUDE_PLUGIN_ROOT.*hardcoded\|hardcoded.*CLAUDE_PLUGIN_ROOT\|CLAUDE_PLUGIN_ROOT.*hardcoded path\|hardcoded path' \
+    "$REPO/lib/12-plugins.sh"
+}
+
+@test "I67: lib/12 hooks.json patch uses sed with _SUBCON_DIR substitution" {
+  grep -q 'sed.*CLAUDE_PLUGIN_ROOT.*_SUBCON_DIR\|sed.*_SUBCON_DIR.*CLAUDE_PLUGIN_ROOT' \
+    "$REPO/lib/12-plugins.sh"
+}
+
+@test "I67: lib/12 hooks.json patch is idempotent (guarded by grep check)" {
+  grep -q "grep.*CLAUDE_PLUGIN_ROOT.*_SUBCON_HOOKS\|grep -q.*CLAUDE_PLUGIN_ROOT.*_SUBCON_HOOKS" \
+    "$REPO/lib/12-plugins.sh"
+}
+
+@test "I67: assembled titan-setup.sh includes CLAUDE_PLUGIN_ROOT hooks.json patch" {
+  grep -q 'CLAUDE_PLUGIN_ROOT.*hooks.json\|hooks.json.*CLAUDE_PLUGIN_ROOT' "$REPO/titan-setup.sh"
+}

--- a/titan-setup.sh
+++ b/titan-setup.sh
@@ -3078,6 +3078,18 @@ else
           warn "subconscious: npm install failed — hooks may not work"
       fi
 
+      # Patch hooks.json: replace ${CLAUDE_PLUGIN_ROOT} with hardcoded absolute path.
+      # CC does not inject CLAUDE_PLUGIN_ROOT into hook subprocesses — empty string causes
+      # "Cannot find module '/hooks/...'" (MODULE_NOT_FOUND) on every hook invocation.
+      # Guard: only apply when literal '${CLAUDE_PLUGIN_ROOT}' is still present (idempotent).
+      _SUBCON_HOOKS="$_SUBCON_DIR/hooks/hooks.json"
+      if [[ -n "$_SUBCON_DIR" ]] && [[ -f "$_SUBCON_HOOKS" ]] &&
+          grep -q '\${CLAUDE_PLUGIN_ROOT}' "$_SUBCON_HOOKS" 2>/dev/null; then
+        sed -i "s|\${CLAUDE_PLUGIN_ROOT}|${_SUBCON_DIR}|g" "$_SUBCON_HOOKS" &&
+          ok "subconscious: patched hooks.json (CLAUDE_PLUGIN_ROOT → hardcoded path)" ||
+          warn "subconscious: hooks.json CLAUDE_PLUGIN_ROOT patch failed"
+      fi
+
       # Patch /dev/tty crash: createWriteStream('/dev/tty') throws unhandled ENXIO in hook context
       # (no terminal). Add .on("error") handler so the stream fails gracefully instead of crashing.
       _SUBCON_SESS="$_SUBCON_DIR/scripts/session_start.ts"

